MekHQ/src/mekhq/campaign/force/StrategicFormation.java

```diff
@@ -600,4 +611,63 @@ public static double calculateTotalWeight(Campaign campaign, int forceId) {
 
         return weight;
     }
+
+    /**
+     * This static method updates the strategic formations across the campaign.
+     * It starts at the top level force, and calculates the strategic formations for each sub-force.
+     * It keeps only the eligible strategic formations and imports them into the campaign.
+     * After every formation is processed, an 'OrganizationChangedEvent' is triggered by that force.
+     *
+     * @param campaign the current campaign.
+     */
+    public static void recalculateStrategicFormations(Campaign campaign) {
```

Comment:
This is resetting all AtB lance roles to Unassigned in my save. (Keeping a copy of the old hash table and passing it into the recursive bit to copy roles from old formation to new fixed it, although there may be a better way.)

Reply:
Thank you so much for catching this! I went ahead and refactored the method so that it's both more efficient and retained pre-existing entries where suitable.